src/goto-instrument/contracts/dynamic-frames/dfcc_dsl_contract_handler.cpp

```diff
@@ -0,0 +1,210 @@
+  const irep_idt &wrapper_id,
+  const irep_idt &wrapped_id,
+  const irep_idt &contract_id,
+  const symbolt &wrapper_write_set_symbol,
+  goto_programt &dest,
+  std::set<irep_idt> &function_pointer_contracts)
+{
+  dfcc_dsl_wrapper_programt(
+    contract_mode,
+    utils.get_function_symbol(wrapper_id),
+    utils.get_function_symbol(wrapped_id),
+    get_contract_functions(contract_id),
+    wrapper_write_set_symbol,
+    goto_model,
+    log,
+    utils,
+    library,
+    instrument)
+    .add_to_dest(dest, function_pointer_contracts);
+}
+
+const symbolt *dfcc_dsl_contract_handlert::get_pure_contract_symbol_ptr(
+  const irep_idt &contract_id)
+{
+  const auto &contract_symbol = utils.get_function_symbol(contract_id);
+  const symbolt *pure_contract_symbol = nullptr;
+  auto pure_contract_id = "contract::" + id2string(contract_id);
+  if(!ns.lookup(pure_contract_id, pure_contract_symbol))
+  {
+    log.debug() << "contract_symbol: " << contract_symbol.name << messaget::eom;
+    log.debug() << contract_symbol.type.pretty() << messaget::eom;
+    log.debug() << "pure_contract_symbol: " << pure_contract_id
+                << messaget::eom;
+    log.debug() << pure_contract_symbol->type.pretty() << messaget::eom;
+
+    check_signature_compat(
+      contract_id,
+      to_code_type(contract_symbol.type),
+      pure_contract_id,
+      to_code_type(pure_contract_symbol->type));
+  }
+  else
+  {
+    // The contract symbol might not have been created if the function had
+    // no contract or a contract with all empty clauses (which is equivalent).
+    // in that case we create a fresh symbol again with empty clauses
+    symbolt new_symbol;
+    new_symbol.name = pure_contract_id;
+    new_symbol.base_name = pure_contract_id;
+    new_symbol.pretty_name = pure_contract_id;
+    new_symbol.is_property = true;
+    new_symbol.type = contract_symbol.type;
+    new_symbol.mode = contract_symbol.mode;
+    new_symbol.module = contract_symbol.module;
+    new_symbol.location = contract_symbol.location;
+    auto entry = goto_model.symbol_table.insert(std::move(new_symbol));
+    if(!entry.second)
+    {
+      log.error().source_location = contract_symbol.location;
+      log.error() << "contract '" << contract_symbol.display_name()
+                  << "' already set at " << entry.first.location.as_string()
+                  << messaget::eom;
+      throw 0;
+    }
+    // this lookup must work, and no need to check for signature compatibility
+    ns.lookup(pure_contract_id, pure_contract_symbol);
+  }
+  return pure_contract_symbol;
+}
+
+const symbolt &dfcc_dsl_contract_handlert::get_pure_contract_symbol(
+  const irep_idt &contract_id)
+{
+  auto result = get_pure_contract_symbol_ptr(contract_id);
+  if(result != nullptr)
+    return *result;
+
+  log.error() << "dfcc_dsl_contract_handlert: pure contract symbol for "
+              << contract_id << " was not found, aborting" << messaget::eom;
+  throw 0;
```

Comment:
Given the above code this should be impossible.

Reply:
fixed